### PR TITLE
Wrote documentation for Str method match

### PR DESCRIPTION
**Pull request by ahalbert**
I wrote documentation for the method Str.match. Once merged, this completes issue #838

### CODE REVIEW

File: doc/Type/Str.pod6 [hunk 1]
```diff
@@ -166,7 +166,58 @@ Examples:
 
     method match($pat, :continue(:$c), :pos(:$p), :global(:$g), :overlap(:$ov), :exhaustive(:$ex), :st(:$nd), :rd(:$th), :$nth, :$x --> Match)
 
-TODO
+Performs a match of the string aginst C<$pat> and returns a L<Match> object if there is a successful match,
```
Comment (titsuki) [inline doc/Type/Str.pod6]: *against

File: doc/Type/Str.pod6 [hunk 2]
```diff
@@ -166,7 +166,58 @@ Examples:
...
+Takes an integer as an argument and returns the nth match in the string. 
+
+=item :x
+
+Takes as an argument the number of matches to return, stopping once the specified number of matches has been reached.
+
+Examples:
+=begin code
+
+"properly".match('perl'); # ｢perl｣
```
Comment (titsuki) [inline doc/Type/Str.pod6]: I think it is better to call `say` like:
```
say "properly".match('perl');
```

File: doc/Type/Str.pod6 (hunk 1 above)
Comment (ahalbert) [inline doc/Type/Str.pod6]: Fixed

Comment (titsuki) [inline doc/Type/Str.pod6]: It seems not yet fixed

Comment (ahalbert) [inline doc/Type/Str.pod6]: I misinterpreted what you said. I fixed the typos

**Comment by titsuki** [comment]
Thanks a lot !

**Pull request merged**